static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -0,0 +1,250 @@
+                        control: (provided: any) => ({
+                          ...provided,
+                          minHeight: '21px',
+                          height: '21px',
+                        }),
+                      }}
+                      inline={false}
+                      isSearchable={false}
+                      isClearable={false}
+                      name={`actionFilters.${id}.logicType`}
+                      required
+                      flexibleControlStateSize
+                      options={FILTER_MATCH_OPTIONS}
+                      size="xs"
+                      value={actionFilterBlock?.logicType}
+                      onChange={value => updateIfLogicType(id, value)}
+                    />
+                  </EmbeddedWrapper>
+                ),
+              })}
+            </StepLead>
+            <DeleteButton
+              aria-label={t('Delete If/Then Block')}
+              size="sm"
+              icon={<IconDelete />}
+              borderless
+              onClick={onDelete}
+            />
+          </Flex>
+          <RuleNodeList
+            placeholder={t('Filter by...')}
+            group={`actionFilters.${id}`}
+            conditions={actionFilterBlock?.conditions || []}
+            onAddRow={type => addIfCondition(id, type)}
+            onDeleteRow={index => removeIfCondition(id, index)}
+            updateCondition={(index, comparison) =>
+              updateIfCondition(id, index, comparison)
+            }
+          />
+        </Flex>
+      </Step>
+      <Step>
+        <StepLead>
+          {tct('[then:Then] perform these actions', {
+            then: <Badge />,
+          })}
+        </StepLead>
+        {/* TODO: add actions dropdown here */}
+      </Step>
+    </IfThenWrapper>
+  );
+}
+
+const PurpleTextButton = styled(Button)`
+  color: ${p => p.theme.purple300};
+  font-weight: normal;
+  padding: 0;
+`;
+
+const Step = styled(Flex)`
+  flex-direction: column;
+  gap: ${space(0.75)};
+`;
+
+const StepLead = styled(Flex)`
+  align-items: center;
+  gap: ${space(0.5)};
+`;
+
+const Badge = styled('span')`
+  display: inline-block;
+  background-color: ${p => p.theme.purple300};
+  padding: 0 ${space(0.75)};
+  border-radius: ${p => p.theme.borderRadius};
+  color: ${p => p.theme.white};
+  text-transform: uppercase;
+  text-align: center;
+  font-size: ${p => p.theme.fontSizeMedium};
+  font-weight: ${p => p.theme.fontWeightBold};
+  line-height: 1.5;
+`;
+
+const EmbeddedSelectField = styled(SelectField)`
+  padding: 0;
+  font-weight: ${p => p.theme.fontWeightNormal};
+  text-transform: none;
+`;
+
+const EmbeddedWrapper = styled('div')`
+  width: 80px;
+`;
+
+const IfThenWrapper = styled(Flex)`
+  flex-direction: column;
+  gap: ${space(1.5)};
+  border: 1px solid ${p => p.theme.border};
+  border-radius: ${p => p.theme.borderRadius};
+  padding: ${space(1.5)};
+  padding-top: ${space(1)};
+  margin-top: ${space(1)};
+`;
```

Comment:
Seems like this might be a generic `Panel` or `Container` component?

Reply:
I feel hesitant about creating another `Panel`/`Container` component for this since we already have so many 😅 I also don't think this particular one would be used anywhere else?

Reply:
Very fair! Consider my comment withdrawn.